### PR TITLE
system log

### PR DESCRIPTION
**Pull request by Frankonly**
#1801

### CODE REVIEW

File: action/protocol/execution/evm/evm.go [hunk 1]
```diff
@@ -45,6 +46,19 @@ func CanTransfer(db vm.StateDB, fromHash common.Address, balance *big.Int) bool
...
+	sdb, ok := db.(*StateDBAdapter)
+	if !ok {
+		log.L().Panic("failed to get StateDBAdapter")
+	}
+	systemLog := systemlog.LogEvmTransfer(&systemlog.EvmTransfer{
+		Amount: amount,
+		From:   outAddr.String(),
+		To:     inAddr.String(),
+	})
+	sdb.AddSystemLog(systemLog)
```
Comment (CoderZhi) [inline action/protocol/execution/evm/evm.go]: Define a global topic
InContractTransfer := make([]byte, 8, 8) // 8 bytes with all 0s
```
db.AddLog(&types.Log{
  Topics: [InContractTransfer, fromHash, toHash],
   Data: []byte{amount},
})
```

Comment (Frankonly) [reply action/protocol/execution/evm/evm.go]: Ok. I was not sure whether there would be a conflict with native topic, so I didn't use zero bytes topic before.

File: action/receipt.go [hunk 2]
```diff
@@ -44,8 +47,11 @@ func (receipt *Receipt) ConvertToReceiptPb() *iotextypes.Receipt {
 	r.GasConsumed = receipt.GasConsumed
 	r.ContractAddress = receipt.ContractAddress
 	r.Logs = []*iotextypes.Log{}
-	for _, log := range receipt.Logs {
-		r.Logs = append(r.Logs, log.ConvertToLogPb())
+	for _, l := range receipt.Logs {
+		// exclude system log when calculating receipts' hash or storing logs
+		if new(big.Int).SetBytes(l.Topics[0][:]).Cmp(big.NewInt(0)) == 1 {
```
Comment (CoderZhi) [inline action/receipt.go]: bytes.Equal(..., ...)

**Comment by CoderZhi** [comment]
Fix CI before merge

**Comment by CoderZhi** [comment]
Add unit tests to cover the log type you introduced

**Pull request merged**